lib/grp.gi

```diff
@@ -130,14 +130,103 @@ end);
 InstallOtherMethod(MinimalGeneratingSet,"fallback method to inform user",true,
   [IsObject],0,
 function(G)
+  local
+    check,GbyGk,
+    Gkm1byGk,
+    Gkm1byGk_L,
+    Gkm1byGk_elem_reps,
+    Gkm1byGk_gen,
+    Gkm1byGk_gen_reps,
```

Comment:
These variably names are... a bit cryptic. That doesn't exactly help understanding the algorithm.

Perhaps you can come up with better names; or at least add a brief comment to each variable indicating what it represents?

Reply:
Added comments.